crates/codegen/runtime/cargo/crate/src/runtime/parser/parser_support/parser_function.rs

```diff
@@ -12,7 +14,7 @@ pub trait ParserFunction<P>
 where
     Self: Fn(&P, &mut ParserContext<'_>) -> ParserResult,
 {
-    fn parse(&self, parser: &P, input: &str) -> ParseOutput;
+    fn parse(&self, parser: &P, input: &str, expected: NonterminalKind) -> ParseOutput;
```

Comment:
nit suggestion, as I think the term is overloaded (expectations in errors):

```suggestion
    fn parse(&self, parser: &P, input: &str, root_kind: NonterminalKind) -> ParseOutput;
```